File: src/main/java/com/bloxmove/marketmaker/model/MarketMakerRequest.java
package com.bloxmove.marketmaker.model;

import lombok.Getter;
import lombok.Setter;

import javax.validation.constraints.DecimalMin;
import javax.validation.constraints.Min;
import javax.validation.constraints.NotNull;
import java.math.BigDecimal;

@Getter
@Setter
public class MarketMakerRequest {

    @NotNull
    ExchangeName exchangeName;
    @NotNull
    String currencyPair;
    @NotNull
    BigDecimal amount;
    @NotNull
    BigDecimal minPrice;
    @NotNull
    BigDecimal targetPrice;
    @NotNull
    BigDecimal maxPrice;
    @NotNull
    @Min(1)
    Long delay;
    @NotNull
    @Min(1)
    Long gridCount;
    @NotNull
    BigDecimal minFirstCurrency;
    @NotNull
    BigDecimal minSecondCurrency;
}

File: market-maker-front/src/app/market-maker/marketMakerRequest.ts
export interface MarketMakerRequest {
    exchangeName: string;
    currencyPair: string;
    amount: number;
    minPrice: number;
    targetPrice: number;
    maxPrice: number;
    delay: number;
    gridCount: number;
    minFirstCurrency: number;
    minSecondCurrency: number;
    needCancelOrders: boolean;
}